lib/exp.py

```diff
@@ -271,12 +271,19 @@ def to_body(self):
 
     def result(self, job):
         status = {}
-        if 'job_parameters' in job.metadata.annotations:
-            status['job_parameters'] = json.loads(
-                job.metadata.annotations['job_parameters'])
+
+        if isinstance(job, client.models.V1Job):
+            job_name = job.metadata.name
+            if 'job_parameters' in job.metadata.annotations:
+                status['job_parameters'] = json.loads(
+                    job.metadata.annotations['job_parameters'])
+        elif isinstance(job, str):
+            job_name = job
```

Comment:
so the one issue with your current implementation is if you pass a string job name, the corresponding Result `job_parameters` aren't going to be set and so may be incorrect if they were actually present.

Instead I think a better solution to #28 is to add a new `get_job()` method to the `Client` class which can leverage [read_namespaced_job](https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/BatchV1Api.md#read_namespaced_job) to return the actual Job object, then in job.py you'd add something like:
```
job = c.get_job(job_name, ns)
```
and then pass `job` into the `exp.result()` call instead.